### DISCUSSION TITLE
`createThumbnail()` method returns a darker image of the texture

### DISCUSSION THREAD

**Discussion by dsaglam94** [google/model-viewer#4441] [Q&A]
I am trying to implement a function where I upload a texture image apply it to the existing model and download an existing texture embedded in the model as an image. I can make an image with an existing method called `createThumbnail` using fixed width and height values. But the problem is it returns a much darker version of the texture. I need the exact copy of the texture.

I checked discussions and googled my problem but couldn't find a solution. Is there a way to get the texture as an image from the model? Maybe there might be a method that I don't know. Thanks a lot!

**Comment by elalish** [comment]
I can't quite understand why you would want to use a screenshot to extract the texture, when you can just read it out of the file? The reason a screenshot doesn't work is that we use physically-based rendering, so the render is a function of both texture and lighting. You can get the texture to come through exactly using the KHR_materials_unlit extension in the glTF.
